src/db-api.ts

```diff
@@ -108,12 +137,20 @@ export function execSQL(sql: string): Promise<any> {
   });
 }
 
-export function execSQLWithParams(
+export function execSQLWithParams(sql: string, params?: Record<string, any>) {
+  if (sql.startsWith('SELECT')) {
```

Comment:
1. `select * from mytable` would be sent to writer pool
2. `with my_temp_view as (select * from my_table) select * from my_view` would be sent to writer pool. We are not using "with" statements yet, but I've used them previously quite a lot. They are very useful with complex queries.

I would write this query other way around. If statement starts with insert, update, delete then use writer. In every other case use reader. This would quickly bring out cases where we wrongly use reader and would make sure we never use writer for reads.